src/cca/scripts/uproxy_server.ts

```diff
@@ -56,25 +37,79 @@ export class UproxyServer implements Server {
   }
 }
 
+// Name by which servers are saved to storage.
+const SERVERS_STORAGE_KEY = 'saved-servers';
+
+// Type of the object placed, in serialised form, in storage.
+interface SavedServers {
```

Comment:
Maybe name it `SavedServerMap`?

I'm not sure you need the extra field. I think you could do:
````
interface SavedServers {
  [id: string]: SavedServer
}
````